.github/workflows/d.yml

```diff
@@ -3,21 +3,59 @@ on: [push, pull_request]
 
 jobs:
   test:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
     name: Dub Test
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-latest]
-        dc: [dmd-2.095.0, ldc-1.20.0]
-
+        os: [ubuntu-20.04, windows-latest]
+        dc:
+          - dmd-2.096.1
+          - dmd-2.095.0
+          - ldc-1.26.0
+          - ldc-1.20.0
+        arch:
+          - x86_64
+        clang:
+          - 10
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
-      - name: Install libclang
-        run: sudo apt-get install libclang-3.9-dev
-      - name: Install D compiler
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3
+          bundler-cache: true
+
+      # Ruby puts this on the path which breaks ldc
+      - name: Rename msys64/link.exe
+        if: startsWith(matrix.os,'windows')
+        shell: pwsh
+        run: mv C:\msys64\usr\bin\link.exe C:\msys64\usr\bin\msys-link.exe
+
```

Comment:
This repo doesn't use any Ruby
```suggestion
```

Reply:
Ha, nice one. I didn't even pay attention to what I was copying.